x/emissions/migrations/v4/migrate_test.go

```diff
@@ -0,0 +1,526 @@
+	cdc := s.emissionsKeeper.GetBinaryCodec()
+
+	blockHeightEnded1 := int64(100)
+	blockHeightEnded2 := int64(200)
+	blockHeightEnded3 := int64(300)
+
+	oldTopics := []oldtypes.Topic{
+		{
+			Id:             1,
+			Creator:        "creator",
+			Metadata:       "metadata",
+			LossMethod:     "lossmethod",
+			EpochLastEnded: 0,
+			EpochLength:    blockHeightEnded1,
+			GroundTruthLag: 10,
+			PNorm:          alloraMath.NewDecFromInt64(3),
+			AlphaRegret:    alloraMath.MustNewDecFromString("0.1"),
+			AllowNegative:  false,
+			Epsilon:        alloraMath.MustNewDecFromString("0.01"),
+			// InitialRegret is being reset to account for NaNs that were previously stored due to insufficient validation
+			InitialRegret:          alloraMath.MustNewDecFromString("11"),
+			WorkerSubmissionWindow: 120,
+		},
+		{
+			Id:                     2,
+			Creator:                "creator2",
+			Metadata:               "metadata2",
+			LossMethod:             "lossmethod2",
+			EpochLastEnded:         0,
+			EpochLength:            blockHeightEnded2,
+			GroundTruthLag:         20,
+			PNorm:                  alloraMath.NewDecFromInt64(3),
+			AlphaRegret:            alloraMath.MustNewDecFromString("0.1"),
+			AllowNegative:          false,
+			Epsilon:                alloraMath.MustNewDecFromString("0.01"),
+			InitialRegret:          alloraMath.MustNewDecFromString("11"),
+			WorkerSubmissionWindow: 120,
+		},
+		{
+			Id:                     3,
+			Creator:                "creator3",
+			Metadata:               "metadata3",
+			LossMethod:             "lossmethod3",
+			EpochLastEnded:         0,
+			EpochLength:            blockHeightEnded3,
+			GroundTruthLag:         30,
+			PNorm:                  alloraMath.NewDecFromInt64(3),
+			AlphaRegret:            alloraMath.MustNewDecFromString("0.1"),
+			AllowNegative:          false,
+			Epsilon:                alloraMath.MustNewDecFromString("0.01"),
+			InitialRegret:          alloraMath.MustNewDecFromString("11"),
+			WorkerSubmissionWindow: 130,
+		},
+	}
+	err := s.emissionsKeeper.AddTopicFeeRevenue(s.ctx, 1, cosmosMath.NewInt(20000))
+	s.Require().NoError(err)
+	err = s.emissionsKeeper.AddTopicFeeRevenue(s.ctx, 2, cosmosMath.NewInt(40000))
+	s.Require().NoError(err)
+	err = s.emissionsKeeper.AddTopicFeeRevenue(s.ctx, 3, cosmosMath.NewInt(60000))
+	s.Require().NoError(err)
+
+	err = s.emissionsKeeper.SetTopicStake(s.ctx, 1, cosmosMath.NewInt(20000))
+	s.Require().NoError(err)
+	err = s.emissionsKeeper.SetTopicStake(s.ctx, 2, cosmosMath.NewInt(40000))
+	s.Require().NoError(err)
+	err = s.emissionsKeeper.SetTopicStake(s.ctx, 3, cosmosMath.NewInt(60000))
+	s.Require().NoError(err)
+
+	topicStore := prefix.NewStore(store, types.TopicsKey)
+	for i, oldTopic := range oldTopics {
+		bz, err := proto.Marshal(&oldTopic)
+		s.Require().NoError(err)
+
+		topicStore.Set([]byte("testKey"+strconv.Itoa(i+1)), bz)
+	}
+
+	err = v3.MigrateTopics(s.ctx, store, cdc, *s.emissionsKeeper)
```

Comment:
Shouldn't we test `v4.MigrateTopics` at here?

Reply:
fixed in #587